components/Staking.tsx

```diff
@@ -0,0 +1,257 @@
+          />
+        </div>
+
+        <p>
+          It will take a few moments for the delegation transaction to be
+          prepared, then Prax will generate a view of your transaction. Verify
+          one of the outputs is to the validator you selected, then click
+          'Approve':
+        </p>
+        <img
+          className="max-w-full"
+          src="/images/delegation_prax.png"
+          alt="Delegation prax"
+        />
+
+        <p>
+          You should see a pop-up in the lower right hand of the page indicating
+          that the transaction was approved!
+        </p>
+
+        <img
+          className="max-w-full"
+          src="/images/delegation_popup.png"
+          alt="Delegation popup"
+        />
+
+        <p>
+          You will receive the delegation token associated with that validator.
+          At a later point, you can undelegate by clicking the 'Undelegate'
+          button to undelegate from that validator, and receive staking tokens.
+        </p>
+      </div>
+
+      {delegationTokens.length === 0 && (
+        <div className="w-full bg-white shadow-md rounded-lg p-4">
+          <div className="flex flex-row gap-3 items-center">
+            <div>Waiting for a staking delegation to occur</div>
+            <div
+              className="animate-spin h-5 w-5 border-2 border-blue-500
+            rounded-full border-t-transparent"
+            />
+          </div>
+        </div>
+      )}
+
+      {delegationTokens.length > 0 && (
+        <div className="flex flex-col items-start">
+          <div
+            className="bg-green-100 border-l-4 border-green-500 text-green-700 p-4 w-full"
+            role="alert"
+          >
+            <p className="font-bold">Staked UM successfully!</p>
+          </div>
+          <div
+            className="flex  px-3
+           items-center gap-5 my-5  bg-gray-700 text-white p-3 w-full"
+          >
+            {delegationTokens?.map((balance) => {
+              const validator = validators?.find((validator) =>
+                (
+                  balance?.balanceView?.valueView
+                    ?.value as ValueView_KnownAssetId
+                )?.metadata?.base?.includes(
+                  bech32mIdentityKey({
+                    ik:
+                      validator?.validator?.identityKey?.ik ?? new Uint8Array(),
+                  }),
+                ),
+              )?.validator;
+              return (
+                <div
+                  className="flex flex-col gap-3 items-start justify-center"
+                  key={balance.toJsonString()}
+                >
+                  <div className="flex items-center gap-3 justify-center">
+                    Successfully staked
+                    <ValueViewComponent
+                      valueView={{
+                        valueView: {
+                          // @ts-expect-error TODO: how do we tell typescript these types are correct?
```

Comment:
suggestion: create an issue to address the use of `@ts-expect-error` in multiple places which indicates unresolved type issues.

Reply:
Addressing this in https://github.com/orgs/prax-wallet/projects/1/views/7?query=sort%3Aupdated-desc+is%3Aopen&pane=issue&itemId=84056389&issue=penumbra-zone%7Cguide%7C38